File: 338-counting-bits/counting-bits.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        ans  = [0]*(n+1)

        for i in range(n+1):
            cnt=0
            x=i
            while i:
                i = i & (i-1)
                cnt+=1
            ans[x]=cnt

        return ans

        

File: 338-counting-bits/counting-bits.go
func countBits(n int) []int {

    arr := make([]int,0, n+1)
    k:=0


    for k<n+1 {
        cnt :=0
        temp:=k
        for temp!=0{
            if (temp&1) == 1 {
                cnt++

            }
            temp=temp>>1
        }
        arr = append(arr, cnt)
        k++
    }
    return arr
    
}